torch/_dynamo/symbolic_convert.py

```diff
@@ -1853,14 +1866,24 @@ def STORE_DEREF(self, inst):
             else:
                 self.output.side_effects.store_cell(cell, val)
         else:
+            maybe_cell = self.symbolic_locals.get(inst.argval)
             if isinstance(
-                self.symbolic_locals.get(inst.argval),
+                maybe_cell,
                 variables.NewCellVariable,
             ):
                 self.output.side_effects.store_cell(
                     self.symbolic_locals[inst.argval], self.pop()
                 )
             else:
+                if (
+                    maybe_cell is not None
+                    and maybe_cell.source.name()
+                    not in self.parent.mutated_closure_cell_contents
+                ):
+                    self.parent.mutated_closure_cell_contents.add(
```

Comment:
I don't think the name here is unique : `counter.__closure__[0].cell_contents`. What distinguishes it from another counter with a closure, if this is persisted across the entire `_compile` invocation ?

Reply:
Do we need this across all compile? or just per frame?

Reply:
I was confused by https://github.com/pytorch/pytorch/blob/c861c33a66c3e49010ce676f0e2c0c2e1cf5ec5d/torch/_dynamo/convert_frame.py#L368 and our new `compile` name, so I guess it's just per frame. But it's not clear to me we don't have the same problem with frame, either.